### PR TITLE
Bump alloy to use `get_receipt` hotfix

### PR DESCRIPTION
**Pull request by yash-atreya**
## Motivation

Use `get_receipt` hotfix from updated alloy version.

## Solution

Bump

### CODE REVIEW

**Review by mattsse — COMMENTED** [review]
there's likely something wrong with script or the send logic of alloy

looks like sending tx no requires an explicit setter for the to field for create txs

File: crates/cast/bin/cmd/call.rs [hunk 1]
```diff
@@ -225,7 +226,7 @@ impl CallArgs {
             }
         };
 
-        req.set_input(data.into());
+        req.set_input::<Bytes>(data.into());
```
Comment (mattsse) [inline crates/cast/bin/cmd/call.rs:229]: I don't think the type hint is required
```suggestion
        req.set_input(data);
```

File: crates/cast/bin/cmd/estimate.rs [hunk 2]
```diff
@@ -121,9 +121,9 @@ impl EstimateArgs {
             }
         };
 
-        req.set_input(data.into());
+        req.set_input::<Bytes>(data.into());
```
Comment (mattsse) [inline crates/cast/bin/cmd/estimate.rs:124]: same

**Review by DaniPopes — APPROVED** [review]

File: crates/cast/bin/tx.rs [hunk 3]
```diff
@@ -56,18 +56,20 @@ pub async fn build_tx<
     let chain = chain.into();
 
     let from = from.into().resolve(provider).await?;
-    let to = if let Some(to) = to { Some(to.into().resolve(provider).await?) } else { None };
+    // TODO: Possible bug here?
```
Comment (mattsse) [inline crates/cast/bin/tx.rs:59]: I think this is correct
```suggestion
```

**Review by mattsse — CHANGES_REQUESTED** [review]
few pedantic nits, can be done in a followup

**Comment by yash-atreya** [comment]
Suggestions addressed in #7701

**Review by Alexis98MG — APPROVED** [review]

**Pull request merged**